jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java

```diff
@@ -332,9 +332,20 @@ private static void deployPersistenceUnit(
         try {
             ValidatorFactory validatorFactory = null;
             final HashMap<String, ValidatorFactory> properties = new HashMap<>();
+
+            CapabilityServiceSupport css = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
             if (!ValidationMode.NONE.equals(pu.getValidationMode())) {
-                // Get the CDI-enabled ValidatorFactory
-                validatorFactory = deploymentUnit.getAttachment(BeanValidationAttachments.VALIDATOR_FACTORY);
+                if (css.hasCapability("org.wildfly.bean-validation")) {
+                    // Get the CDI-enabled ValidatorFactory
+                    validatorFactory = deploymentUnit.getAttachment(BeanValidationAttachments.VALIDATOR_FACTORY);
+                }
+                if (validatorFactory == null && ValidationMode.CALLBACK.equals(pu.getValidationMode())) {
+                    // Should we error here? ValidationMode.CALLBACK javadoc says a failure
+                    // should happen if there is no BV available, but perhaps that happens elsewhere?
+                    // In any case the hasCapability check above that accompanied this comment
```

Comment:
From the current JPA 2.2 spec, it sounds like the persistence provider would do any validation, I don't think we need the null check:

3.6.1.1 Enabling Automatic Validation

The validation-mode element of the persistence.xml file determines whether the automatic
lifecycle event validation is in effect. The values of the validation-mode element are AUTO,
CALLBACK, NONE. The default validation mode is AUTO.

If the application creates the entity manager factory using the Persistence.createEntityMan-
agerFactory method, the validation mode can be specified using the javax.persis-
tence.validation.mode map key, which will override the value specified (or defaulted) in the
persistence.xml file. The map values for this key are "auto", "callback", "none".

If the auto validation mode is specified by the validation-mode element or the javax.persis-
tence.validation.mode property, or if neither the validation-mode element nor the
javax.persistence.validation.mode property is specified, and a Bean Validation provider
is present in the environment, the persistence provider must perform the automatic validation of entities
as described in section 3.6.1.2. If no Bean Validation provider is present in the environment, no lifecy-
cle event validation takes place.

If the callback validation mode is specified by the validation-mode element or the javax.per-
sistence.validation.mode property, the persistence provider must perform the lifecycle event
validation as described in section 3.6.1.2. It is an error if there is no Bean Validation provider present in
the environment, and the provider must throw the PersistenceException if the javax.per-
sistence.validation.mode property value "callback" has been passed to the Persis-
tence.createEntityManagerFactory method.

If the none validation mode is specified by the validation-mode element or the javax.per-
sistence.validation.mode property, the persistence provider must not perform lifecycle event
validation.
"

Reply:
Other than the unneeded null check, I think the rest is fine.